### DISCUSSION TITLE
List all easy OT parameters list

### DISCUSSION THREAD

**Discussion by bwired-nl** [Jeroen88/EasyOpenTherm#6] [Q&A] [answered]
Hi
Great, mqtt operational👍
Is there a list with all the paraneters we can use.
For example hvac waterpressure etc

**Comment by Jeroen88** [comment, marked as answer]
> Hi Great, mqtt operational👍 Is there a list with all the paraneters we can use. For example hvac waterpressure etc

There are only a few DATA-IDs mandatory, for a boiler these are: ```STATUS, SECONDARY_CONFIGURATION```, ```RELATIVE_MODULATION_LEVEL```, ```BOILER_WATER_TEMP```, ```CONTROL_SETPOINT_CH``` and ```MAX_MODULATION_LEVEL```. For a HVAC I think this suggests that ```HVAC_STATUS``` will be mandatory too. So all others are _optional_.

Just try all ```OpenTherm::READ_DATA_ID```'s and test with ```OpenTherm::error()``` the response. If the response is ```OpenTherm::ERROR_CODES::UNKNOWN_DATA_ID``` your HVAC does not support the DATA-ID.

**Reply by Jeroen88** [reply]
I added an example, OpenTherm_Show_DATA-IDs,ino, that tries all ```READ_DATA_ID```s and shows the resulting response. The example can also be used to understand the meaning of the flags better. Finally it shows the right units (if any). I hope it will help you!

**Reply by bwired-nl** [reply]
got it, my arduino IDe was old :-(  but we need to install some extra libs, like the pubsub etc, there are a lot of them available:-)
My knowledge is also not that good yet :-)
Thanks
